src/Symfony/Component/BrowserKit/Tests/AbstractBrowserTest.php

```diff
@@ -60,6 +60,17 @@ public function testXmlHttpRequest()
         $this->assertFalse($client->getServerParameter('HTTP_X_REQUESTED_WITH', false));
     }
 
+    public function testJsonRequest()
+    {
+        $client = $this->getBrowser();
+        $client->jsonRequest('GET', 'http://example.com/', ['param' => 1], [], true);
+        $this->assertEquals($client->getRequest()->getServer()['CONTENT_TYPE'], 'application/json');
+        $this->assertEquals($client->getRequest()->getServer()['HTTP_ACCEPT'], 'application/json');
```

Comment:
arguments should be swapped ("expected" is always first)
+ assertSame in all the method

Reply:
To avoid problems when merging 4.4 to 5.x I created a separate PR for the other tests: #38646